webapp/src/main/java/org/apache/atlas/web/rest/AuthREST.java

```diff
@@ -136,4 +150,43 @@ public ServicePolicies downloadPolicies(@PathParam("serviceName") final String s
             AtlasPerfTracer.log(perf);
         }
     }
+
+    private boolean isPolicyUpdated(String serviceName, long lastUpdatedTime) {
+        AtlasPerfMetrics.MetricRecorder recorder = RequestContext.get().startMetricRecord("CachePolicyTransformerImpl.isPolicyUpdated" + serviceName);
+
+        AuditSearchParams parameters = new AuditSearchParams();
+        Map<String, Object> dsl = getMap("size", 1);
+
+        List<Map<String, Object>> mustClauseList = new ArrayList<>();
+        mustClauseList.add(getMap("term", getMap("typeName", POLICY_ENTITY_TYPE)));
+
+        lastUpdatedTime = lastUpdatedTime == -1 ? 0 : lastUpdatedTime;
+        mustClauseList.add(getMap("range", getMap("timestamp", getMap("gte", lastUpdatedTime))));
+
+        dsl.put("query", getMap("bool", getMap("must", mustClauseList)));
+
+        parameters.setDsl(dsl);
+
+        try {
+            EntityAuditSearchResult result = auditRepository.searchEvents(parameters.getQueryString());
+
+            if (result == null || CollectionUtils.isEmpty(result.getEntityAudits())) {
+                LOG.info("getPoliciesIfUpdated: Skipping as no update found");
+                return false;
+            }
+        } catch (AtlasBaseException e) {
+            LOG.error("ERROR in getPoliciesIfUpdated while fetching entity audits {}: ", e.getMessage());
+            return true;
+        } finally {
+            RequestContext.get().endMetricRecord(recorder);
+        }
+
+        return true;
+    }
+
+    private Map<String, Object> getMap(String key, Object value) {
```

Comment:
nit:: Since this is query, we can always use ImmutableMap.of(K,V) so that it's not modified down stream.

Reply:
actually, this map or parameters is not passed to the auditRepository, instead we extract DSL query & send it as a String to auditRepository